stdlib/Logging/src/ConsoleLogger.jl

```diff
@@ -58,11 +58,16 @@ end
 showvalue(io, ex::Exception) = showerror(io, ex)
 
 function default_logcolor(level::LogLevel)
-    level in keys(custom_log_levels) ? custom_log_levels[level][2] :
-    level < Info  ? Base.debug_color() :
-    level < Warn  ? Base.info_color()  :
-    level < Error ? Base.warn_color()  :
-                    Base.error_color()
+    color = get(_log_levels, level, nothing)
+    if isnothing(color)
+        # default colors: based such that colors for standard log levels (Info etc) are in middle of color range
+        level = level.level
+        color = level < -500 ? Base.debug_color() :
+                level <  500 ? Base.info_color()  :
+                level < 1500 ? Base.warn_color()  :
+                               Base.error_color()
+    end
+    return color
```

Comment:
This logic is here, which I am unsure about.
It could instead be inside the code for `@create_log_macro`
to be called at definition time.

Pros of in putting it in `@create_log_macro`:
 -  is that the color wouldn't be Logger specific, so user configured loggers could take advantage of it.
 
Cons of in putting it in `@create_log_macro`:
 - We would need some trick to make it actually continue to read the enviroment variable behimds `debug_color` etc at runtime (rather than at compile time). I am not sure what that trick would be.
  - Wouldn't have a definition for what color to use if the log level was defined via the constructor only, rather than the macro. Which people do do right now. Could have the logic in both, or simpler logic that just says use color `1` if we do not have a color defined for that level.